src/rpc/RPCHelpers.cpp

```diff
@@ -1391,4 +1395,51 @@ toJsonWithBinaryTx(data::TransactionAndMetadata const& txnPlusMeta, std::uint32_
     return obj;
 }
 
+std::expected<std::set<std::pair<ripple::AccountID, ripple::Slice>>, Status>
+createAuthCredsByCredentialID(
+    std::optional<boost::json::array> const& credID,
+    std::shared_ptr<BackendInterface> const& backend,
+    ripple::LedgerHeader const& info,
+    Context const& ctx
+)
+{
+    std::set<std::pair<ripple::AccountID, ripple::Slice>> authCreds;
+    if (credID.value().size() > ripple::maxCredentialsArraySize) {
+        return Error{Status{RippledError::rpcINVALID_PARAMS, "credential array too long."}};
+    }
+
+    for (auto const& elem : credID.value()) {
+        ASSERT(elem.is_string(), "should already be checked in validators.hpp that elem is a string.");
+
+        ripple::uint256 credHash;
+        ASSERT(
+            credHash.parseHex(boost::json::value_to<std::string>(elem)),
+            "should already be checked in validators.hpp that elem is a uint256 hex"
+        );
+
+        auto const credKeylet = ripple::keylet::credential(credHash).key;
+        auto const credLedgerObject = backend->fetchLedgerObject(credKeylet, info.seq, ctx.yield);
+        if (!credLedgerObject)
+            return Error{Status{RippledError::rpcBAD_CREDENTIALS, "credentials aren't accepted."}};
+
+        auto credIt = ripple::SerialIter{credLedgerObject->data(), credLedgerObject->size()};
+        auto sleCred = ripple::SLE{credIt, credKeylet};
+
+        if (!credLedgerObject || (sleCred.getType() != ripple::ltCREDENTIAL) ||
+            ((sleCred.getFieldU32(ripple::sfFlags) & ripple::lsfAccepted) == 0u))
+            return Error{Status{RippledError::rpcBAD_CREDENTIALS}};
+
+        if (credentials::checkExpired(sleCred, info))
+            return Error{Status{RippledError::rpcBAD_CREDENTIALS}};
+
+        auto const [it, ins] = authCreds.insert(
+            {sleCred.getAccountID(ripple::sfIssuer), ripple::makeSlice(sleCred.getFieldVL(ripple::sfCredentialType))}
```

Comment:
slice does not hold the memory. Is it okay? `sleCred.getFieldVL(ripple::sfCredentialType)` returns a temporary  Blob, slice will point to invalid memory.